gtsfm/densify/patchmatchnet_data.py

```diff
@@ -0,0 +1,259 @@
+"""Interface class from GtsfmData to PatchmatchNetData
+
+Authors: Ren Liu
+"""
+from typing import Any, Dict, List, Tuple
+
+import cv2
+import numpy as np
+from torch.utils.data import Dataset
+
+from gtsfm.common.image import Image
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.densify.mvs_math import piecewise_gaussian
+
+NUM_PATCHMATCHNET_STAGES = 4
+
+
+class PatchmatchNetData(Dataset):
+    """PatchmatchNetData class for PatchmatchNet. It contains the interface from GtsfmData.
+    Ref: Wang et al. https://github.com/FangjinhuaWang/PatchmatchNet/blob/main/datasets/dtu_yao_eval.py
+    """
+
+    def __init__(self, images: Dict[int, Image], sfm_result: GtsfmData, num_views: int = 5) -> None:
+        """Initialize method for PatchmatchNetData
```

Comment:
Thanks for all the updates to the PR, Ren.

I think the docstring here could be more informative.

how about something like
‘’’
Cache image and camera pose metadata for PatchmatchNet inference.
‘’’
And then you could add an inline comment explaining
# store locations of camera centers in the world frame

# set image dimensions to those of zero’th image

# create mapping from image indices (with some entries potentially missing) to [0,N-1] tensor indices for N inputs to PatchmatchNet